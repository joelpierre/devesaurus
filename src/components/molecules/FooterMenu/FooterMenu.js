import React from 'react';
import { graphql, StaticQuery } from 'gatsby';
import PropTypes from 'prop-types';

import styles from './FooterMenu.module.scss';

import MenuItem from '../../atoms/MenuItem/MenuItem';

export const PureFooterMenu = ({
  className,
  allWordpressWpApiMenusMenusItems,
}) => (
  <nav
    data-test="component-footer-menu"
    className={`${styles['footer-menu']} ${className}`}
  >
    <ul className={styles['footer-menu__list']}>
      {allWordpressWpApiMenusMenusItems.edges[0].node.items.map(item => (
        <MenuItem
          className={styles['footer-menu__item']}
          key={item.order}
          item={item}
        />
      ))}
    </ul>
  </nav>
);

const FooterMenu = ({ className }) => (
  <StaticQuery
    data-test="component-footer-menu-query"
    query={menuQuery}
    render={props => <PureFooterMenu {...props} className={className} />}
  />
);

FooterMenu.defaultProps = {
  className: null,
};

FooterMenu.propTypes = {
  className: PropTypes.string,
};

export default FooterMenu;

const menuQuery = graphql`
  {
    allWordpressWpApiMenusMenusItems(filter: { name: { eq: "Footer Menu" } }) {
      edges {
        node {
          name
          items {
            order
            title
            object_slug
          }
        }
      }
    }
  }
`;
