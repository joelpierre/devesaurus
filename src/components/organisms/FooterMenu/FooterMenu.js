import React from 'react';
import PropTypes from 'prop-types';

const FooterMenu = (props) => {
  return (<></>);
};

FooterMenu.defaultProps = {};

FooterMenu.propTypes = {};

export default FooterMenu;
