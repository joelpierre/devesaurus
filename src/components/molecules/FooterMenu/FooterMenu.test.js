import React from 'react';
import { shallow } from 'enzyme';
import { StaticQuery } from 'gatsby';
import toJson from 'enzyme-to-json';

import {
  checkProps,
  findByTestAttr,
  matchSnapshot,
} from '../../../utils/test-utilities';

import FooterMenu, { PureFooterMenu } from './FooterMenu';

const defaultProps = {};

const defaultPureProps = {
  allWordpressWpApiMenusMenusItems: {
    edges: [
      {
        node: {
          items: [
            {
              title: 'title',
              object_slug: 'slug',
            },
          ],
        },
      },
    ],
  },
};

/**
 * Factory function to create a ShallowWrapper for the FooterMenu component.
 * @param {object} props - Component props specific to setup
 * @returns {ShallowWrapper}
 */
const setup = (props = {}) => {
  const setupProps = { ...defaultProps, ...props };
  return shallow(<FooterMenu {...setupProps} />);
};

const pureSetup = (props = {}) => {
  const setupProps = { ...defaultPureProps, ...props };
  return shallow(<PureFooterMenu {...setupProps} />);
};

describe('<FooterMenu/>', () => {
  let wrapper;
  let pureWrapper;

  beforeEach(() => {
    wrapper = setup();
    pureWrapper = pureSetup();
  });

  it('renders the FooterMenu Component without errors query', () => {
    const component = findByTestAttr(wrapper, 'component-footer-menu-query');
    expect(component.length).toBe(1);
  });

  it('renders the FooterMenu Component without errors', () => {
    const component = findByTestAttr(pureWrapper, 'component-footer-menu');
    expect(component.length).toBe(1);
  });

  it('should render with correct props', () => {
    checkProps(FooterMenu, defaultProps);
  });

  it('should render with correct props', () => {
    checkProps(PureFooterMenu, defaultProps);
  });

  it('should match snapshot', () => {
    matchSnapshot(wrapper);
  });

  it('should match snapshot', () => {
    matchSnapshot(pureWrapper);
  });
});
