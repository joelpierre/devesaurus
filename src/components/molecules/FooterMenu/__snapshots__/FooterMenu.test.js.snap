// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FooterMenu/> should match snapshot 1`] = `
<mockConstructor
  data-test="component-footer-menu-query"
  render={[Function]}
/>
`;

exports[`<FooterMenu/> should match snapshot 2`] = `
<nav
  className="footer-menu undefined"
  data-test="component-footer-menu"
>
  <ul
    className="footer-menu__list"
  >
    <MenuItem
      className="footer-menu__item"
      item={
        Object {
          "attr": "attr",
          "classes": "test-class",
          "object_slug": "slug",
          "order": "1",
          "title": "title",
          "url": "url",
        }
      }
      key="1"
    />
  </ul>
</nav>
`;
